Work with methods without red mistakes.

udodikdmitro opened the pull request.

Commit:
Work with methods without red mistakes.

Commit:
New approach.

udodikdmitro commented on src/package1/CustomArrayList.java:
```diff
@@ -4,104 +4,216 @@
 
     @Override
-    public boolean isEmpty(List<String> name) {
+    public int size() {
+        int i = 0;
+
+        for (String s : elements) {
+            i++;
+        }
+        return i;
+    }
```
I'll use variable size instead

udodikdmitro replied:
Done

Commit:
Only some methods and go working in a big car.

ezseeton commented on src/package1/CustomArrayList.java:
```diff
@@ -5,55 +5,46 @@
-            if (s != null) {
-                return false;
-            }
+        if (size == 0) {
+            return true;
         }
```
use return with condition instead

udodikdmitro replied:
Changed everywhere.

ezseeton commented on src/package1/CustomArrayList.java:
```diff
@@ -33,75 +40,162 @@ public boolean contains(List<String> name, Object o) {
-        String [] a = new String[name.size() + 1];
+    public boolean add(String s) {
+
+        elements[size] = s;
+        size++;
 
-        for (int i = 0; i < name.size(); i++) {
-            a[i] = name.get(i);
+        if (size == elements.length) {
+            String[] a = new String[(int) (elements.length * 1.5)];
+            elements = a;
         }
```
forgot to copy previous elements to created array

udodikdmitro replied:
Now it copies previous elements.

ezseeton commented on src/package1/CustomArrayList.java:
```diff
@@ -33,75 +40,162 @@ public boolean contains(List<String> name, Object o) {
-        for (int i = 0; i < name.size(); i++) {
-            if (name.get(i).equals(o)) {
-                name.remove(name.get(i));
+    public boolean remove(Object o) {
```
1. Use size, not length
2. Remove is just shift elements to left

udodikdmitro replied:
Fixed.

ezseeton commented on src/package1/CustomArrayList.java:
```diff
@@ -33,75 +40,162 @@ public boolean contains(List<String> name, Object o) {
+
+        for (Object s : c) {
 
+            for (int i = 0; i < elements.length; i++) {
```
size

udodikdmitro replied:
Now iterator() is using.

ezseeton commented on src/package1/CustomArrayList.java:
```diff
@@ -33,75 +40,162 @@ public boolean contains(List<String> name, Object o) {
-            for (String s : c) {
+    public boolean containsAll(Collection<?> c) {
+
+        int j = 0;
```
use meaningful names of variables. For example "amountOfMatchs" instead of j

udodikdmitro replied:
Changes names.

ezseeton reviewed (changes requested):
Please use size everywhere in code. Use meaningful variable names

Commit:
Methods are overed except the last method. Some methods don't work.

Commit:
- use size() everywhere
- implement method iterator() and create CustomArrayListIterator for it
- add new constructor to create instance with specified capacity
- use iterator in contains() and containsAll() methods
- fix logic of remove() and add() methods
